indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java

```diff
@@ -366,33 +372,35 @@ public String getVersion(final Interval interval)
     finally {
       if (normalExit) {
         try {
-          // Always want to persist.
-          log.info("Persisting remaining data.");
-
-          final Committer committer = committerSupplier.get();
-          final CountDownLatch persistLatch = new CountDownLatch(1);
-          plumber.persist(
-              new Committer()
-              {
-                @Override
-                public Object getMetadata()
-                {
-                  return committer.getMetadata();
-                }
-
-                @Override
-                public void run()
+          // Persist if we had actually started.
+          if (firehose != null) {
+            log.info("Persisting remaining data.");
+
+            final Committer committer = committerSupplier.get();
+            final CountDownLatch persistLatch = new CountDownLatch(1);
+            plumber.persist(
```

Comment:
At this point `finishingJob` is false and `firehose` is not null, so if the firehose is drainable by closing (like the event receiver) then nothing special will happen, the persist will finish.